app/components/TimerButton.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react'
+import { View, Text, StyleSheet } from 'react-native'
+import config, { colors, vectorIcons } from '../config'
+import { AnimatedCircularProgress } from 'react-native-circular-progress'
+import { useDispatch } from 'react-redux'
+import { addWashTime } from '../state/WashTimeHistory'
+
+const TimerButton = ({ navigation }) => {
+  const dispatch = useDispatch()
+  const timerDefault = config.timerDefault
+  const [timer, setCounter] = useState(timerDefault)
+  const [text, setText] = useState('Seconds')
+
+  const timerCounter = () => setCounter(timer - 1)
+
+  useEffect(() => {
+    if (timer <= 0) {
+      setText('Well Done')
+      dispatch(addWashTime({ datetime: Date.now() }))
+      return
+    }
+    const id = setInterval(timerCounter, 1000)
+    return () => {
+      clearInterval(id)
+    }
+  }, [timer])
+
+  const fill = 100 - timer * 5
+
+  const { FontAwesome5 } = vectorIcons
+  const { white } = colors
+  const {
+    container,
+    textSeconds,
+    textSecondName,
+    viewStyle,
+    progressStyle,
+  } = styles
+
+  const renderTimer = () => {
+    return (
+      <View>
+        {timer === 0 ? (
+          <FontAwesome5 name={'thumbs-up'} size={30} color={white} />
+        ) : (
+          <Text style={textSeconds}>{timer}</Text>
+        )}
+      </View>
+    )
+  }
```

Comment:
Why are we doing a function like this? It should be in the main return or split out. I think this is small enough to put in the return?

Also this button should be reused on the home screen:

<img width="899" alt="Screenshot 2020-05-01 at 08 45 28" src="https://user-images.githubusercontent.com/875335/80790957-26251900-8b88-11ea-9c61-d34881c65b9a.png">

That is why you need the TouchableOpcity on it. You would also need to make the icon here generic (was can pass in a name) and the text value changes.

Reply:
At homeScreen we just need to show image. there are no progress needed and no state requires for the same. so that we use image at home screen and Circular progress on Timer screen.

Reply:
I disagree. The home screen has an icon and text the same, same styling/spacing. I think we can make this one component.

Reply:
Ok i have use same component for both Home Screen and Timer Screen by using props.